ECK Resources Helm Chart (#5781)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/stack-helm-chart.asciidoc

[source, sh]
----
helm repo add elastic https://helm.elastic.co
helm repo update
----

NOTE: The minimum supported version of Helm is 3.2.0.

NOTE: ECK Stack Helm Charts are currently being released as an Enterprise licensed feature.

[float]
[id="{p}-install-elasticsearch-kibana-helm"]
== Installing Elasticsearch and Kibana using the eck-stack Helm Chart

Similar to the <<{p}-deploy-elasticsearch,quickstart>>, the following section describes how to setup an Elasticsearch cluster with a simple Kibana instance managed by ECK, and how to customize a deployment using the eck-stack Helm chart's values.

[source,sh]
----
# Install an eck-managed Elasticsearch and Kibana using the default values, which deploys the quickstart examples.
helm install es-kb-quickstart elastic/eck-stack -n elastic-stack --create-namespace
----

[float]
[id="{p}-eck-stack-helm-customize"]
=== Customizing Kibana and Elasticsearch using the eck-stack Helm Chart's example values

> There are many example Helm values files for installing and managing a more advanced Elasticsearch and/or Kibana link:{eck_github}/tree/{eck_release_branch}/deploy/eck-stack/examples[in the project repository].

Reviewer: ```suggestion
There are example Helm values files for installing and managing a more advanced Elasticsearch and/or Kibana link:{eck_github}/tree/{eck_release_branch}/deploy/eck-stack/examples[in the project repository].
```

[source, sh]
----
helm repo add elastic https://helm.elastic.co
helm repo update
----

NOTE: The minimum supported version of Helm is 3.2.0.

[float]
[id="{p}-install-elasticsearch-helm"]
== Installing Elasticsearch using the eck-stack Helm Chart

Similar to the <<{p}-deploy-elasticsearch,quickstart>>, the following section describes how to setup a simple Elasticsearch cluster specification, with one Elasticsearch node, managed by ECK, and how to customize an Elasticsearch deployment using the eck-stack Helm chart.

[source,sh]
----
# Install an eck-managed Elasticsearch cluster using the default values, which deploys the quickstart example.
helm install es-quickstart elastic/eck-stack -n elastic-stack --create-namespace
----

[float]
[id="{p}-elasticsearch-helm-customize"]
=== Customizing Elasticsearch using eck-stack Helm Chart

There are many example Helm values files for installing and managing a more advanced Elasticsearch cluster link:{eck_github}/tree/{eck_release_branch}/deploy/eck-stack/examples/elasticsearch[in the project repository].

To use one of these example configurations, use the `--values` Helm option, as seen in the following section.

[source,sh]
----

> # Install an eck-managed Elasticsearch cluster using the node roles example with hot, warm, and cold node groups.

Reviewer: ```suggestion
# Install an eck-managed Elasticsearch cluster using the node roles example with hot, warm, and cold data tiers.
```
I think that's the official term.

helm install es-quickstart elastic/eck-stack -n elastic-stack --create-namespace --values https://github.com/elastic/cloud-on-k8s/tree/main/deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml
----

[float]
[id="{p}-install-kibana-helm"]
== Installing Kibana using the eck-stack Helm Chart

> Similar to the <<{p}-deploy-kibana,quickstart>>, the following section describe how to setup a simple Kibana instance, along with a simple Elasticsearch cluster managed by ECK, and how to customize a Kibana deployment using the eck-stack Helm chart.

Reviewer: ```suggestion
Similar to the <<{p}-deploy-kibana,quickstart>>, the following section describes how to setup a simple Kibana instance, along with a simple Elasticsearch cluster managed by ECK, and how to customize a Kibana deployment using the eck-stack Helm chart.
```

[source,sh]
----
# Install an eck-managed Elasticsearch and Kibana using the default values, which deploys the quickstart example.

> helm install es-kb-quickstart elastic/eck-stack -n elastic-stack --create-namespace --set eck-kibana.enabled=true

Reviewer: Should the default be that Kibana is always enabled?

Author: This has been changed, and documentation updated.

----

[float]
[id="{p}-kibana-helm-customize"]
=== Customizing Kibana using eck-stack Helm Chart

There are many example Helm values files for installing and managing a more advanced Kibana instance link:{eck_github}/tree/{eck_release_branch}/deploy/eck-stack/examples/kibana[in the project repository].

To use one of these example configurations, use the `--values` Helm option, as seen in the following section.

[source,sh]
----

> # Install an eck-managed Elasticsearch and Kibana using the Elasticsearch node roles example with hot, warm, and cold node groups, and the Kibana example customizing the http service.

Reviewer: ```suggestion
# Install an eck-managed Elasticsearch and Kibana using the Elasticsearch node roles example with hot, warm, and cold data tiers, and the Kibana example customizing the http service.
```

:page_id: stack-helm-chart
ifdef::env-github[]
****
link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View this document on the Elastic website]
****
endif::[]
[id="{p}-{page_id}"]
= Elastic Stack Helm Chart

Starting from ECK 2.4.0, a Helm chart is available for managing Elastic Stack resources using the ECK Operator. It is available from the Elastic Helm repository and can be added to your Helm repository list by running the following command:

[source, sh]
----
helm repo add elastic https://helm.elastic.co
helm repo update
----

NOTE: The minimum supported version of Helm is 3.2.0.

> NOTE: ECK Stack Helm Charts are currently being released as an Enterprise-level licensing feature.

Reviewer: ```suggestion
NOTE: ECK Stack Helm Charts are currently being released as an Enterprise licensed feature.
```

[float]
[id="{p}-install-elasticsearch-helm"]
== Installing Elasticsearch using the eck-stack Helm Chart

Similar to the <<{p}-deploy-elasticsearch,quickstart>>, the following section describes how to setup a simple Elasticsearch cluster specification, with one Elasticsearch node, managed by ECK, and how to customize an Elasticsearch deployment using the eck-stack Helm chart.

[source,sh]
----
# Install an eck-managed Elasticsearch cluster using the default values, which deploys the quickstart example.
helm install es-quickstart elastic/eck-stack -n elastic-stack --create-namespace --set eck-kibana.enabled=false
----

[float]
[id="{p}-elasticsearch-helm-customize"]

> === Customizing Elasticsearch using eck-stack Helm Chart

Reviewer: ```suggestion
=== Customizing Elasticsearch using the eck-stack Helm Chart
```

To use one of these example configurations, use the `--values` Helm option, as seen in the following section.

[source,sh]
----
# Install an eck-managed Elasticsearch cluster using the node roles example with hot, warm, and cold data tiers.
helm install es-quickstart elastic/eck-stack -n elastic-stack --create-namespace --values https://github.com/elastic/cloud-on-k8s/tree/main/deploy/eck-stack/examples/elasticsearch/hot-warm-cold.yaml --set eck-kibana.enabled=false
----

[float]
[id="{p}-install-kibana-helm"]
== Installing Kibana using the eck-stack Helm Chart

Similar to the <<{p}-deploy-kibana,quickstart>>, the following section describes how to setup a simple Kibana instance, along with a simple Elasticsearch cluster managed by ECK, and how to customize a Kibana deployment using the eck-stack Helm chart.

[source,sh]
----
# Install an eck-managed Elasticsearch and Kibana using the default values, which deploys the quickstart examples.
helm install es-kb-quickstart elastic/eck-stack -n elastic-stack --create-namespace
----

[float]
[id="{p}-kibana-helm-customize"]

> === Customizing Kibana using eck-stack Helm Chart

Reviewer: ```suggestion
=== Customizing Kibana using the eck-stack Helm Chart
```

> There are many example Helm values files for installing and managing a more advanced Kibana instance link:{eck_github}/tree/{eck_release_branch}/deploy/eck-stack/examples/kibana[in the project repository].

Reviewer: This almost verbatim repeats the Elasticsearch section. I wonder if a better structure for this page would be:
* Installing the Elastic stack using the eck-stack Helm chart (describes the default case Kibana+ES)
* Customising the Elastic stack using the eck-stack Helm chart (describes how to use the values files)
* Installing individual stack applications (describes how to turn off for example Kibana and/or how to use the sub- charts directly)

Author: Most recent update reorganizes this documentation @pebrc .
